src/Model/UserItem.php

```diff
@@ -207,20 +207,22 @@ private static function getProfileForUser(int $uid)
 	 */
 	private static function checkShared(array $item, int $uid)
 	{
+		// Only sheck on starting posts and reshare ("announce") activities, otherwise return
 		if (($item['gravity'] != GRAVITY_PARENT) && ($item['verb'] != Activity::ANNOUNCE)) {
 			return false;
 		}
 
-		// Either the contact had posted something directly
+		// Check if the contact had posted or shared something directly
 		if (DBA::exists('contact', ['id' => $item['contact-id'], 'notify_new_posts' => true])) {
 			return true;
 		}
 
-		if ($item['gravity'] != GRAVITY_PARENT) {
+		// Don't continue when the item had been an announce activity
```

Comment:
Yes but why?

Reply:
Because the following test then doesn't make any sense.

Reply:
Thanks, can you please add this to the comment in the code as well?

Reply:
Not today. I'm too tired, it's too hot and my system currently holds some uncommitted stuff that I possibly need to clean first being able doing so.

Reply:
It's okay, it's isn't a rush, take your time, cool down.

Reply:
Done. Over night temperature went down to 28°C (from 32 in the evening), so I had a little bit more energy :-)